spec/unit/indirector/file_bucket_file/file_spec.rb

```diff
@@ -53,10 +53,36 @@ def save_bucket_file(contents, path = "/who_cares")
                               0x6d658673,0xa4341f7d,0x8fd75920,0xefd18d5a].pack("I" * 16)
 
         save_bucket_file(first_contents, "/foo/bar")
-
         expect do
           save_bucket_file(collision_contents, "/foo/bar")
-        end.to raise_error(Puppet::FileBucket::BucketError, /Got passed new contents/)
+        end.to raise_error(Puppet::FileBucket::BucketError, /Existing backup and new file have different content but same checksum.*(?!#{Puppet[:bucketdir]})/)
+      end
+
+      # See PUP-1334
+      context "when the contents file exists but is corrupted and does not match the expected checksum" do
+        let(:original_contents) { "a file that will get corrupted" }
+        let(:bucket_file) { Puppet::FileBucket::File.new(original_contents) }
+        let(:contents_file) { "#{Puppet[:bucketdir]}/8/e/6/4/f/8/5/d/8e64f85dd54a412f65edabcafe44d491/contents" }
```

Comment:
Yes, this should be getting cleaned up per https://github.com/puppetlabs/puppet/blob/master/spec/spec_helper.rb#L155-L156